File: rPi-music-box-things/src/main/java/com/christiankula/rpimusicbox/things/injection/modules/NearbyModule.java
package com.christiankula.rpimusicbox.things.injection.modules;

import android.content.Context;

import com.google.android.gms.nearby.Nearby;
import com.google.android.gms.nearby.connection.AdvertisingOptions;
import com.google.android.gms.nearby.connection.ConnectionsClient;
import com.google.android.gms.nearby.connection.Strategy;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class NearbyModule {

    @Provides
    @Singleton
    ConnectionsClient provideConnectionsClient(Context context) {
        return Nearby.getConnectionsClient(context);
    }

    @Provides
    @Singleton
    AdvertisingOptions provideAdvertisingOptions(){
        return new AdvertisingOptions(Strategy.P2P_STAR);
    }
}

File: rPi-music-box-remote/src/main/java/com/christiankula/rpimusicbox/remote/nearby/injection/NearbyModule.kt
package com.christiankula.rpimusicbox.remote.nearby.injection

import android.content.Context
import com.christiankula.rpimusicbox.nearby.RxNearby
import dagger.Module
import dagger.Provides

@Module
class NearbyModule {

    @Provides
    internal fun provideRxNearby(context: Context): RxNearby {
        return RxNearby(context)
    }
}
